plugins/vendor/zoom/ZoomDropFolderPlugin/batch/processors/zoomRecordingProcessor.php

```diff
@@ -323,11 +339,88 @@ protected function getAdditionalUsers($recordingId, $userToExclude)
 		return $this->getValidatedUsers($additionalZoomUsers, $this->dropFolder->partnerId, $this->dropFolder->zoomVendorIntegration->createUserIfNotExist,
 		                                $userToExclude);
 	}
-	
+
+	/**
+	 * @param string $recordingId
+	 * @param string $userToExclude
+	 * @return array|null
+	 */
+	protected function getValidatedHosts($recordingId, $userToExclude)
+	{
+		$userToExclude = strtolower($userToExclude);
+		$zoomCoHosts = $this->getCoHostsEmails($recordingId);
+		$zoomAlternativeHosts = $this->getAlternativeHostsEmails($recordingId);
+		$zoomHosts = $zoomCoHosts ? array_merge($zoomCoHosts, $zoomAlternativeHosts) : $zoomAlternativeHosts;
+		$zoomHosts = $zoomHosts ? $zoomHosts : null;
+		return $this->getValidatedUsers($zoomHosts, $this->dropFolder->partnerId, $this->dropFolder->zoomVendorIntegration->createUserIfNotExist,
+										$userToExclude);
+	}
+
+	protected function getAlternativeHostsEmails($recordingId)
+	{
+		$data = $this->getAlternativeHostsData($recordingId);
+		if(isset($data[self::SETTINGS]) && isset($data[self::SETTINGS][self::ALTERNATIVE_HOSTS]))
+		{
+			KalturaLog::debug("Found the following alternative hosts [" . $data[self::SETTINGS][self::ALTERNATIVE_HOSTS] . "]");
+			$alternativeHostsEmails = explode(";", $data[self::SETTINGS][self::ALTERNATIVE_HOSTS]);
+			return $this->parseZoomEmails($alternativeHostsEmails);
+		}
+		return array();
+	}
+
+	protected function getCoHostsEmails($recordingId)
+	{
+		$coHostsEmails = array();
+		$pageIndex = 0;
+		$maxPages = 10;
+		$maxPageSize = 300;
+		$nextPageToken = '';
+		do
+		{
+			$metricsParticipants = $this->getCoHostsData($recordingId, $maxPageSize, $nextPageToken);
+			$participants = new kZoomParticipants();
+			$participants->parseData($metricsParticipants, self::COHOST_ROLE);
+			$coHostsEmails = array_merge($coHostsEmails, $participants->getParticipantsEmails());
+			$pageIndex++;
+			$nextPageToken = $metricsParticipants && $metricsParticipants[self::NEXT_PAGE_TOKEN] ?
+				$metricsParticipants[self::NEXT_PAGE_TOKEN] : '';
+
+		} while ($nextPageToken !== '' && $pageIndex < $maxPages);
+
+		KalturaLog::debug("Found the following co-hosts " . print_r($coHostsEmails, true));
+		return $this->parseZoomEmails($coHostsEmails);
+	}
+
+	protected function parseZoomEmails($emails)
+	{
+		$result = array();
+		if($emails)
+		{
+			$emails = array_filter(array_unique($emails), 'trim');
+			KalturaLog::debug('Found the following emails: ' . implode(", ", $emails));
+			foreach ($emails as $email)
+			{
+				$zoomUser = new kZoomUser();
+				$zoomUser->setOriginalName($email);
+				$zoomUser->setProcessedName(ZoomBatchUtils::processZoomUserName($email, $this->dropFolder->zoomVendorIntegration, $this->zoomClient));
+				$result[] = $zoomUser;
+			}
+		}
+		else
+		{
+			$result = null;

```

Comment:
on webinar we used to return array() while now we will return now, need to make sure all calling functions follow it equivalently